Update table documentation

eerovaher opened the pull request:
<!-- This comments are hidden when you submit the pull request,
so you do not need to remove them! -->

<!-- Please be sure to check out our contributing guidelines,
https://github.com/astropy/astropy/blob/main/CONTRIBUTING.md .
Please be sure to check out our code of conduct,
https://github.com/astropy/astropy/blob/main/CODE_OF_CONDUCT.md . -->

<!-- If you are new or need to be re-acquainted with Astropy
contributing workflow, please see
http://docs.astropy.org/en/latest/development/workflow/development_workflow.html .
There is even a practical example at
https://docs.astropy.org/en/latest/development/workflow/git_edit_workflow_examples.html#astropy-fix-example . -->

<!-- Astropy coding style guidelines can be found here:
https://docs.astropy.org/en/latest/development/codeguide.html#coding-style-conventions
Our testing infrastructure enforces to follow a subset of the PEP8 to be
followed. You can check locally whether your changes have followed these by
running the following command:

tox -e codestyle

-->

<!-- Please just have a quick search on GitHub to see if a similar
pull request has already been posted.
We have old closed pull requests that might provide useful code or ideas
that directly tie in with your pull request. -->

<!-- We have several automatic features that run when a pull request is open.
They can appear daunting but do not worry because maintainers will help
you navigate them, if necessary. -->

### Description
<!-- Provide a general description of what your pull request does.
Complete the following sentence and add relevant details as you see fit. -->

<!-- In addition please ensure that the pull request title is descriptive
and allows maintainers to infer the applicable subpackage(s). -->

Many of the changes add hyperlinks or make sure that function names get rendered as `func()` instead of `func`, but there are a few more substantial changes. See the commit messages for details.

Table docs also contain a handful of code examples that have caused problems on Windows and are therefore skipped in doctests. I've re-enabled those tests so that CI could run here because that's the only way I can test the code on Windows. If those tests are still causing problems for Windows then I'll have to redo the last commit of this PR.

### Checklist for package maintainer(s)
<!-- This section is to be filled by package maintainer(s) who will
review this pull request. -->

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [x] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [x] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label.
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

eerovaher commented:
Most of the tests I had re-enabled succeeded, but one still failed on Windows and the 32-bit platform, so I replaced the last commit.

eerovaher commented:
I tried using `.. doctest-skip:: win32` and that allowed the tests to succeed on Windows, but not on 32-bit platform, so it looks like that one test still has to be skipped entirely.

nstarman reviewed (commented):
Overall this looks great and like a lot of work!! I didn't have time to go through everything, so if other maintainers could take a look, that would be great. I noted a few places where I think the RST links appear shorter but are both longer and less readable when the docs are compiled. Also, any insight by @astropy/documentation-infrastructure-maintainers on ``FLOAT_CMP`` and comparison precision would be useful in keeping some of the table examples readable and now doc-tested.

eerovaher commented on docs/table/operations.rst:
```diff
@@ -268,27 +268,28 @@ Aggregation
+  >>> print(obs_mean)  # doctest: +FLOAT_CMP
+  name       mag_b              mag_v
+  ---- ------------------ ------------------
+  M101 15.000000000000002 13.725000000000001
```
The problem with tables in doctests is that the number of digits in the output determines the number of dashes in the line preceding the numerical values. So 
```
  name    mag_b
  ---- -----------
  M101        15.0
```
would succeed, but 
```
  name mag_b
  ---- -----
  M101  15.0
```
fails. `+FLOAT_CMP` is working as intended.

I don't think the first example here looks any better than what is in the docs right now.

nstarman replied:
Thanks for the explanation. Good to know ``+FLOAT_CMP`` is working.
To me the 
```
  name    mag_b
  ---- -----------
  M101        15.0
```
is better for two reasons. The first is readability. Having some spaces doesn't detract as much as 14 extraneous digits. The second is correctness. ``15.0`` is the correct number. Presumably a different computer architecture (e.g. big-endian or 32 bit) would have different computer error. ``+FLOAT_CMP``  prevents including computer error into the doctest.

eerovaher replied:
Setting column formats is the best solution.

nstarman replied:
@taldcroft? resolve because #12022 will address this point.

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -128,28 +128,28 @@ You can get summary information about the table as follows::
-     c  8.0 4.24264068712   2  14
+  name mean        std        min max
+  ---- ---- ----------------- --- ---
+     a  6.0 4.242640687119285   0  12
```
See https://github.com/astropy/astropy/pull/12011#discussion_r682524353

eerovaher replied:
Formatting the output of `Table.info('stat')` would be simpler, but it doesn't seem to be possible to do that right now. I propose we keep the changes I've already made so that this example could be tested, and deal with the formatting in #12022 instead.

taldcroft replied:
You can use offset variations on the array `[2, 4, 4, 4, 5, 5, 7, 9] ` (found it on wikipedia). This has a mean of 5.0 and a std of 2.0. Then the info table will always be looking tidy.

eerovaher replied:
This would work, but replacing the initial array would mean that many more examples would have to be replaced, not just the ones with `info('stats')`. On the other hand that might also be the case if we choose to address this in #12022 and also choose to set the format of the other columns.

What do others think?

pllim replied:
I think we're getting ourselves into a bike shed. As it is, this PR is already a great improvement. Though "ugly," the extra decimal points do reflect real life cases that doctest can run. I am okay with deferring pretty output to #12022 if others think that is the proper way to go.

taldcroft replied:
My only real worry is if these tests will pass on a weird architecture or a 32-bit machine (remembering that the problem is in the number of dashes, not the float compare). As long as that is true then I don't mind the example looking messy.

taldcroft replied:
And I guess a related comment is that the original version of the docs represents how it used to look since they were definitely copy-pasted from running the commands interactively. So the formatting has changed in the meantime.

pllim replied:
If the floating points change that much in 32-bit, can we just use ellipsis and call it a day? Is that an option?

taldcroft replied:
The problem is in the header -- the number of spaces in the column row and the number of dashes will change if the floating point representation changes by a digit. So ellipsis won't help. I'm happy enough to revert back to skipping the test, and I'd prefer that to the possibility that the doctest will fail under some circumstances.

eerovaher replied:
When #12022 is merged then it will become possible to fully control the number of digits in the printed output and therefore the number of spaces and dashes in the table header.

nstarman replied:
As that PR is proving more challenging than initially estimated and this PR will probably be merged first, are there alternatives for now?

taldcroft replied:
I would propose to revert the changes and go back to skipping these tests. Perfect is the enemy of good. Apart from this one thing I'm very happy with this PR (and also greatly appreciate the patience of @eerovaher to stick with it!).

eerovaher replied:
The difficult part in #12022 was figuring out what the output here should be, but after the discussion there it looks like the best way forward is to apply better default formatting instead of applying column formats. Implementing that is quite simple and the only reason I haven't finalized #12022 already is because if it gets merged first then it would create a conflict with this PR. So the output here can be made better-looking and it can be tested too, the only open question still left is the order in which those two updates should be applied. Merging this PR as is and then changing the appearance of the output in #12022 looks like a simpler way to proceed.

taldcroft replied:
OK, let's just merge this as-is and go ahead with #12022, where you rebase on main. That will cause this test to immediately start failing and you can fix it then.

@nstarman - are all your concerns resolved, or close enough?

eerovaher replied:
@taldcroft, all the tests succeed here, why would there be any failures after merging?

nstarman replied:
@eerovaher's plan is a good one. Since there's a roadmap to reformat the example, I'm all good here!

Re. to failing tests, I think @taldcroft means this docs example would fail if in #12022 you update the table format, but don't change this example. This example can't slip under the radar and not be fixed.

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -458,16 +455,14 @@ supported navigation keys are:
-arguments ``max_lines``, ``max_width``, ``show_name``, ``show_unit`` with
-meanings as shown below::
+In order to fully control the print output use the
+:meth:`astropy.table.Table.pprint` or :meth:`astropy.table.Column.pprint`
```
This is a question of style preference, so it makes sense to adopt your suggestion if enough people support it. Unless someone more familiar with Sphinx knows a way to have these links show up as `Table.pprint()` etc because that would be clearly preferred.

pllim replied:
I am neutral on this, so I'll defer to @taldcroft or @mhvk as Table maintainers.

nstarman replied:
Resolved with a 3rd solution. Thanks @eerovaher!

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -439,10 +436,10 @@ To print a formatted table::
-In order to browse all rows of a table or column use the Table
-:meth:`~astropy.table.Table.more` or Column :func:`~astropy.table.Column.more`
+In order to browse all rows of a table or column use the
+:meth:`astropy.table.Table.more` or :meth:`astropy.table.Column.more`
```
See https://github.com/astropy/astropy/pull/12011#discussion_r682533421

nstarman replied:
@eerovaher's solution in the latest commits is better than the above options.
Marking as resolved.

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -481,14 +476,13 @@ meanings as shown below::
-           ... ...          ...
-  2.940000e+03 ...       2969.0
-  2.970000e+03 ...       2999.0
+  >>> t.pprint(max_lines=8, max_width=40, show_unit=False)
```
`show_unit=True` is the default behavior, so there is hardly any reason to use it at all.

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -613,12 +607,12 @@ pformat() method
-:func:`~astropy.table.Column.pformat` methods. These behave just as for
-:meth:`~astropy.table.Table.pprint` but return a list corresponding to each
-formatted line in the :meth:`~astropy.table.Table.pprint` output. The
+the :meth:`astropy.table.Table.pformat` or :meth:`astropy.table.Column.pformat`
```
See https://github.com/astropy/astropy/pull/12011#discussion_r682533421

eerovaher commented on docs/table/access_table.rst:
```diff
@@ -904,7 +893,9 @@ The examples below illustrate dealing with bytestring data in ``astropy``.
-    array([ True, False], dtype=bool)
+    array([ True, False])
+
+.. doctest-skip::
```
Previously this example was skipped by the `doctest-skip-all` directive at line 879, so replacing that with a `doctest-skip` here increases test coverage by allowing the preceding examples to be tested.

The example deals with I/O, so if this was to be tested it would be necessary to add setup and teardown code, and it also requires `h5py` to be installed. It certainly looks easier to skip this test.

pllim commented:
Amazing! Doc clean-ups are usually not the work people tend to want to do, so thank you so much for tackling this! I'll provide a full review soon, but in the meantime...

Re: https://github.com/astropy/astropy/pull/12011#issuecomment-892260177 -- I have encountered this 64 vs 32 bit problem in the past. I got around it by using the `#doctest: +ELLIPSIS` trick (though I don't think it works with `FLOAT_CMP`, so you have to make a choice there).

nstarman commented on docs/table/operations.rst:
```diff
@@ -394,19 +397,19 @@ key columns::
   ...     colnames = [name for name in table.colnames if name not in key_colnames]
   ...     for colname in colnames:
-  ...         if np.any(table[colname] < 0):
+  ...         if np.any(table[colname] <= 0):
```
good catch.

nstarman commented on docs/table/pandas.rst:
```diff
@@ -56,21 +55,18 @@ It is also possible to create a table from a `DataFrame`_::
-* Tables with mixin columns can currently not be converted, but this may be
-  implemented in the future.
-
-.. _DataFrame: http://pandas-docs.github.io/pandas-docs-travis/
```
great to delete this. 👍

nstarman commented on docs/table/operations.rst:
```diff
@@ -302,6 +303,8 @@ to perform the aggregation::
                 17.0
   15.699999999999998
```
Depending on how the conversation about ``+FLOAT_CMP`` resolves, this might be good to change as well.

```suggestion
                15.7
```

nstarman replied:
Done!

nstarman commented on docs/table/operations.rst:
```diff
@@ -268,27 +268,28 @@ Aggregation
+   M82 15.699999999999998               15.5
+
+It seems the magnitude values were successfully averaged, but what about the
+``AstropyUserWarning``? Since the ``obs_date`` column is a string-type array,
```
Maybe a full link?

```suggestion
:class:`~astropy.utils.exceptions.AstropyUserWarning`? Since the ``obs_date`` column is a string-type array,
```

nstarman replied:
Implemented.

nstarman reviewed (commented):
Finished my first look. Overall this is really great! Re-request a review when you get a chance to look at some of suggestions.

pllim commented on docs/table/performance.inc.rst:
```diff
@@ -3,6 +3,8 @@
    from === to ***, the filename extension needs to be changed from .inc.rst to
    .rst, and a link needs to be added in the subpackage toctree
 
+.. doctest-skip-all
```
Why skip all here? I thought the original intention was to test some of the stuff here, but maybe subpackage maintainers can confirm.

eerovaher replied:
Most of the examples here were excluded from tests individually anyways, and the couple that weren't did not contain any example output to be tested. That made me think that skipping all the tests wouldn't cause test coverage to drop much at all.

pllim reviewed (commented):
Overall, nice clean-up and it is good to have more of the examples actually tested. Thank you! 👏 

I left a general review but subpackage maintainers should also have a look, as there are some content changes, in addition to stylistic.

Nitpick: Inconsistent use of `:class:` throughout; while this does not affect the rendered output, would still be nice to be consistent.

I am also going to run the cron jobs here, as I see new links added. Want to make sure the new doctests and links don't break stuff.

For other reviewers, rendered doc can be viewed at https://astropy--12011.org.readthedocs.build/en/12011/

eerovaher commented on docs/table/construct_table.rst:
```diff
@@ -227,9 +227,6 @@ need to use the ``rows`` keyword to create a table::
     4 5.0   y
     5 8.2   z
 
-The data object passed as the ``rows`` argument can be any form which is
```
This sentence is outdated since #9165.

eerovaher commented on docs/table/construct_table.rst:
```diff
@@ -658,52 +674,45 @@ names
-be copied with the standard library ``copy.deepcopy()`` routine. By default,
-``meta`` is an empty :class:`~collections.OrderedDict`.
+be copied with the standard library :func:`copy.deepcopy()` routine. By
+default, ``meta`` is an empty :class:`~collections.OrderedDict`.
```
It is indeed the plan to remove `OrderedDict` from `astropy`, but you can see from #10888 that most subpackages including `astropy.table` haven't done it yet. So removing the references to `OrderedDict` from the documentation would be premature.

pllim replied:
Ah, okay. Thanks for all the clarifications!

nstarman replied:
Marking as resolved. OrderedDict lives to see another PR.

eerovaher commented on docs/table/construct_table.rst:
```diff
@@ -715,44 +724,43 @@ The `Row data`_ section provides details.
-  >>> from astropy.table import QTable
   >>> dat = [[1, 2], ['hello', 'world']]
-  >>> qt = QTable(dat, names=['a', 'b'], units=(u.m, None))
+  >>> t = Table(dat, names=['a', 'b'], units=(u.m, None))
```
The point of the change was to demonstrate that units can be used with both `Table` and `QTable`, but it could be reverted if that's preferable.

pllim replied:
I think switching it from one line to the next without any text to explain why is too confusing. Also I think the general recommendation is to always use `QTable` when you have to deal with units.

nstarman replied:
I think this is reverted. @pllim, resolve conversation?

pllim replied:
> resolve conversation?

For future reference, feel free to click resolve if you think it is resolved. You did not need my permission. 😉

eerovaher commented on docs/table/index.rst:
```diff
@@ -169,7 +170,7 @@ syntax::
   4
 
-  >>> t[1]         # Row object for table row index=1
+  >>> t[1]         # Row 1 of the table
```
Zero-based indexing is what Python uses, and the example output should clear any confusion.

eerovaher commented on docs/table/index.rst:
```diff
@@ -228,16 +229,17 @@ value is given in ``cm / s`` but will be added to the table as ``0.1 m / s`` in
-  >>> t = QTable([a, b, c], names=('a', 'b', 'c'), masked=True, dtype=('i4', 'f8', 'U1'))
-  >>> t['a'].mask = [True, True, False]
+  >>> from astropy.table import MaskedColumn
+  >>> a_masked = MaskedColumn(a, mask=[True, True, False])
```
The documentation updates in #8789 did include such a recommendation.

eerovaher commented on docs/table/masking.rst:
```diff
@@ -109,9 +113,6 @@ a |MaskedColumn|.
       1     3
       2     4
 
-Notice the table attributes ``mask`` and ``fill_value`` that are
```
Those attributes do exist and are described later in the document, but they are not included in any of the printed output so there is nothing to notice here.

eerovaher commented on docs/table/mixin_columns.rst:
```diff
@@ -194,41 +194,21 @@ Adding or inserting a row works as expected only for mixin classes that are
   ValueError: Unable to insert row because of exception in column 'skycoord':
   'SkyCoord' object has no attribute 'insert'
 
-**Initializing from a list of rows or a list of dicts**
```
This was fixed by #9165.

eerovaher commented on docs/table/operations.rst:
```diff
@@ -19,28 +19,28 @@ a new table from one or more input tables. This includes:
    * - `Join`_
      - Database-style join of two tables
-     - `~astropy.table.join`
+     - |join|
```
`|join|` was already defined in the first line of the file, so it made sense to use it.

pllim replied:
Ah, okay!

eerovaher commented:
> Nitpick: Inconsistent use of `:class:` throughout; while this does not affect the rendered output, would still be nice to be consistent.

I agree that it would be better to be more consistent with the use of `:class:`, but at this point ensuring that would require quite a lot of work and it wouldn't even affect the rendered HTML.

pllim commented:
Re: `:class:` -- I don't mean for you to go clean up all the docs, but at least be consistent in your diff? I don't see why we need to switch between both of these in the same patch:

```
:class:`numpy.<func>`

`numpy.<func>`
```

taldcroft commented on docs/table/pandas.rst:
```diff
@@ -56,21 +55,18 @@ It is also possible to create a table from a `DataFrame`_::
+  columns, so |Table| objects with multidimensional columns cannot be converted
+  to :class:`~pandas.DataFrame`.
+
+* Masked tables can be converted, but :class:`~pandas.DataFrame` uses
```
Actually this is no longer a problem for int dtype since #9541 and support for pandas nullable int. So you should update this bit accordingly.

nstarman replied:
I think this is implemented. @taldcroft, resolve conversation?

taldcroft reviewed (commented):
Amazing work @eerovaher, I cannot say thank you enough!!  I made a few comments and weighed in on a couple of outstanding questions. If you address those then I think this will be OK to approve and merge.

eerovaher commented:
I've addressed the feedback and added a few more hyperlinks.

I also decided to add code examples to the known problems described in the documentation so that if those problems are ever fixed the examples would cause the doctests to fail. There were a handful of such described issues, but I was only able to add a single new code example because it turned out the other problems had been fixed already. See the commit messages for details.

eerovaher commented:
I was able to make Sphinx render the links to `Table` and `Column` formatting methods in the preferred way described in https://github.com/astropy/astropy/pull/12011#discussion_r682533421, so that's what the last commit implements.

nstarman reviewed (commented):
Thanks for tackling all the suggestions. In particular the example table formatting is much improved.

pllim reviewed (commented):
I have left a general review before, so I'll not review again. Final approval should come from subpackage maintainers. Thanks for your continued effort and patience!

nstarman commented on docs/table/access_table.rst:
```diff
@@ -806,8 +803,8 @@ changes to the :attr:`~astropy.table.Column.quantity` property::
 Even without explicit conversion, columns with units can be treated like
-like an ``astropy`` `~astropy.units.Quantity` in *some* arithmetic
-expressions (see the warning below for caveats to this)::
+like a |Quantity| in *some* arithmetic expressions (see the warning below for
```
Resolved.

eerovaher commented on docs/table/index.rst:
```diff
@@ -99,15 +99,16 @@ You can get summary information about the table as follows::
-``astropy.table.default_notebook_table_class`` configuration item):
+From within a `Jupyter notebook <https://jupyter.org/>`_, the table is
+displayed as a formatted HTML table (details of how it appears can be changed
+by altering the ``astropy.table.conf.default_notebook_table_class``
```
The way configuration works in `astropy` is that the `__init__.py` files of the subpackages define a `Conf` class and then create an instance of that class called `conf`. It would be possible to create a link to the attribute of the `Conf` class, but the users should be using the corresponding `conf` variable instead. I'm not sure if creating the link here would be helpful or confusing.

nstarman replied:
I'd lean towards helpful. We link to class methods and attributes elsewhere and hopefully most users understand python's paradigm where unless it's explicitly stated, e.g. a ``classmethod``, things on classes should be accessed on an instance of the class, not the class itself.

eerovaher replied:
This is a bit different because the user is not expected to create new instances of the `Conf` class, they are only ever supposed to use the hardcoded `conf` variable. But if you still think it is a good idea to create the link then I can do it when I squash the commits.

nstarman replied:
Thanks!

eerovaher commented on docs/table/modify_table.rst:
```diff
@@ -304,18 +314,18 @@ It is possible to force an in-place update of a column as follows::
-replaced. This option must be set to a list that includes zero or more of the
-following string values:
+In order to find potential problems related to replacing columns, there is the
+option ``table.conf.replace_warnings`` in the :ref:`astropy_config`. This
```
See https://github.com/astropy/astropy/pull/12011#discussion_r690578625

nstarman commented on docs/table/access_table.rst:
```diff
@@ -458,16 +455,14 @@ supported navigation keys are:
+In order to fully control the print output use the
+:meth:`astropy.table.Table.pprint` or :meth:`astropy.table.Column.pprint`
+methods. These have keyword arguments ``max_lines``, ``max_width``,
+``show_name``, ``show_unit`` with meanings as shown below::
```
Done.

nstarman commented on docs/table/construct_table.rst:
```diff
@@ -561,29 +578,29 @@ Initialization Details
 object with the following arguments, all of which are optional:
 
-``data`` : numpy ndarray, dict, list, Table, or table-like object, optional
+``data`` : :class:`numpy.ndarray`, `dict`, `list`, |Table|, or table-like object, optional
```
Resolved using ``|ndarray|`` instead.

nstarman commented on docs/table/construct_table.rst:
```diff
@@ -658,52 +674,45 @@ names
-same length as the number of columns. Any list elements with value
-``None`` fall back to the default name.
+override the existing ones. By default, the column names are either taken from
+existing names (for :class:`~numpy.ndarray` or |Table| input) or auto-generated
```
All instances of ``` ``ndarray`` ``` are replaced with ``|ndarray|``

nstarman commented on docs/table/construct_table.rst:
```diff
@@ -658,52 +674,45 @@ names
 :class:`~collections.OrderedDict`, but the only firm requirement is that it can
-be copied with the standard library ``copy.deepcopy()`` routine. By default,
-``meta`` is an empty :class:`~collections.OrderedDict`.
+be copied with the standard library :func:`copy.deepcopy()` routine. By
```
Implemented.

nstarman commented on docs/table/index.rst:
```diff
@@ -99,15 +99,16 @@ You can get summary information about the table as follows::
-``astropy.table.default_notebook_table_class`` configuration item):
+From within a `Jupyter notebook <https://jupyter.org/>`_, the table is
+displayed as a formatted HTML table (details of how it appears can be changed
+by altering the ``astropy.table.default_notebook_table_class`` configuration
```
Implemented. See https://github.com/astropy/astropy/pull/12011/files#r689938483 for followup discussion.

nstarman commented on docs/table/construct_table.rst:
```diff
@@ -294,28 +308,29 @@ the difference between a scalar ``1`` (length 0) and an array such as
       4     5.0    y
 
-Note that in the above example and most the following examples we are creating a
+Note that in the above example and most following examples we are creating a
```
Implemented.

nstarman commented:
Thanks again @eerovaher for the diligent follow through. As this PR is winding up, I've started resolved conversations, checking that the comments were addressed. I think there's only 1 or 2 still active!

taldcroft commented:
@eerovaher - this is good now and I'm planning to approve and merge. Just one last request at the end of this long road, and that is to squash the commits down to one. I think there is not much value-added in the individual commits for these doc fixes.

eerovaher commented on docs/table/index.rst:
```diff
@@ -99,15 +99,17 @@ You can get summary information about the table as follows::
+displayed as a formatted HTML table (details of how it appears can be changed
+by altering the `astropy.table.conf.default_notebook_table_class
+<astropy.table.Conf.default_notebook_table_class>` item in the
+:ref:`astropy_config`:
```
@nstarman, is this what you had in mind?

nstarman replied:
Exactly! Thanks.

eerovaher commented on docs/table/modify_table.rst:
```diff
@@ -304,18 +314,19 @@ It is possible to force an in-place update of a column as follows::
+<astropy.table.Conf.replace_warnings>` in the :ref:`astropy_config`. This
+controls a set of warnings that are emitted under certain circumstances when a
+table column is replaced. This option must be set to a list that includes zero
+or more of the following string values:
```
I've added a link to the attribute in this section too.

eerovaher commented:
I have squashed the commits and made the two changes that @nstarman requested. The cron failures are caused by `apt-get install` and are unrelated to the PR itself.

taldcroft commented:
Have you looked into the ci/circleci: image-tests-mpldev fail? Presumably unrelated?

eerovaher commented:
This is the error message: `ERROR: InvocationError for command /root/project/.tox/py37-test-image-mpldev/bin/pytest --pyargs astropy /root/project/docs --mpl -P visualization --remote-data=astropy --open-files --mpl-results-path=/root/project/results -W ignore:np.asscalar (exited with code 3)`
I'm not sure what it means but it looks like the error occurred before any tests could run at all.

taldcroft reviewed (approved):
Thanks @eerovaher !
